roles/importer/files/importer/ciscoasa9/asa_parser.py

```diff
@@ -0,0 +1,313 @@
+            timeout = int(m.group(2))
+            # look ahead for matching counter line for same module
+            # (could come in any order; we'll scan nearby couple of lines)
+            keepalive_counter = None
+            # Attempt to find a matching counter in the next 3 lines
+            for j in range(i + 1, min(i + 5, len(lines))):
+                mm = re.match(rf"^service-module\s+{re.escape(name)}\s+keepalive-counter\s+(\d+)$",
+                              lines[j].strip(), re.I)
+                if mm:
+                    keepalive_counter = int(mm.group(1))
+                    break
+            if keepalive_counter is None:
+                # fallback: 0
+                keepalive_counter = 0
+            service_modules.append(AsaServiceModule(name=name, keepalive_timeout=timeout, keepalive_counter=keepalive_counter))
+            i += 1
+            continue
+        m = re.match(r"^service-module\s+(\S+)\s+keepalive-counter\s+(\d+)$", line, re.I)
+        if m:
+            # already handled (or no matching timeout yet) – we’ll defer adding if not present
+            i += 1
+            continue
+
+        # name (alias) lines
+        m = re.match(r"^name\s+(\d{1,3}(?:\.\d{1,3}){3})\s+(\S+)(?:\s+description\s+(.+))?$", line, re.I)
+        if m:
+            ip, alias, desc = m.group(1), m.group(2), m.group(3)
+            names.append(Names(name=alias, ip_address=ip, description=desc.strip() if desc else None))
+            i += 1
+            continue
+
+        # interface block
+        if re.match(r"^interface\s+\S+", line, re.I):
+            block, i = _consume_block(lines, i, re.compile(r"^interface\s+\S+", re.I))
+            interfaces.append(_parse_interface_block(block))
+            continue
+
+        # object network block
+        if re.match(r"^object\s+network\s+\S+$", line, re.I):
+            block, i = _consume_block(lines, i, re.compile(r"^object\s+network\s+\S+$", re.I))
+            net_obj, pending_nat = _parse_network_object_block(block)
+            if net_obj:
+                net_objects.append(net_obj)
+            if pending_nat:
+                nat_rules.append(pending_nat)
+            continue
+
+        # object-group network
+        if re.match(r"^object-group\s+network\s+\S+$", line, re.I):
+            block, i = _consume_block(lines, i, re.compile(r"^object-group\s+network\s+\S+$", re.I))
+            net_obj_groups.append(_parse_network_object_group_block(block))
+            continue
+
+        # object service
+        if re.match(r"^object\s+service\s+\S+$", line, re.I):
+            block, i = _consume_block(lines, i, re.compile(r"^object\s+service\s+\S+$", re.I))
+            svc_obj = _parse_service_object_block(block)
+            if svc_obj:
+                svc_objects.append(svc_obj)
+            continue
+
+        # object service with inline protocol definition
+        if re.match(r"^object service \S+ (tcp|udp|icmp|ip|tcp-udp)( .+)?$", line, re.I):
```

Comment:
I guess gre (and possibly other protocols, see https://en.wikipedia.org/wiki/List_of_IP_protocol_numbers?wprov=sfla1) has to be added here